src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1068,8 +1073,30 @@ MonodroidRuntime::convert_dl_flags (int flags)
 	return lflags;
 }
 
-inline void*
-MonodroidRuntime::monodroid_dlopen_log_and_return (void *handle, char **err, const char *full_name, bool free_memory)
+force_inline void
+MonodroidRuntime::init_internal_api_dso (void *handle)
+{
+	if (handle == nullptr) {
+		log_fatal (LOG_DEFAULT, "Internal API library is required");
+		exit (FATAL_EXIT_MONO_MISSING_SYMBOLS);
+	}
+
+	auto api = new MonoAndroidInternalCalls_Impl ();
```

Comment:
The "memory leak" here should be explicitly called out in the commit message, and how we're fine with it.

Reply:
If I'm reading this right, the [idea on how to slightly limit the memory leak](https://discordapp.com/channels/732297728826277939/732297837953679412/736269643232182322) is to only incur the memory leak on Android 9.0 targets which have the dlopen "bug" in the first place, as per above `if (handle == api_dso_handle)` check.

While this does limit things -- and in a manner which doesn't explicitly check the API Level, which is nice if/when a vendor fixes the dlopen issue on their own fork -- it doesn't help *on Android 9 targets*.  Thus, Android 9 targets will still have a leak, and the constraints of this leak should be described.

Reply:
The latest push should have plugged the leak entirely. I wanted to avoid this kind of "heavy" work, but with it limited to few devices and with the mutex held anyway, I think it's OK